### DISCUSSION TITLE
js I want to change the class of all elements with the same classes.

### DISCUSSION THREAD

**Discussion by qSHiFTp** [community/community#132144] [Programming Help] [answered]
### Body

The code is needed to change the site theme

`let body = document.querySelector('body');
let themeButton = document.querySelector('.button_2');
let button1 = document.querySelectorAll('.button_1');
let text = document.querySelectorAll('.text');

function allCl (fname, classname)
{
    for (let i = 0; i < fname.length ; i++) 
        {
            fname.classList.toggle(classname);
            console.log(classname);
        }
}

const darkTheme = window.matchMedia("(prefers-color-scheme: dark)");
if (darkTheme.matches) 
{
    body.classList.toggle( 'dark-theme' );   // Theme set to dark.
    allCl( button1, 'button_1' );
    allCl( text, 'dark-text' );
}
else 
{
    body.classList.toggle( 'light-theme' );   // Theme set to light.
    allCl( button1, 'button_1' );
    allCl( text, 'text' );
}

themeButton.onclick = function () 
{
    body.classList.toggle('light-theme');
    body.classList.toggle('dark-theme');
    allCl( button1, 'button_1' );
    allCl( button1, 'dark-button_1');
    allCl( text, 'text' );
    allCl( text, 'dark-text' );
}`

I'm a newbie, don't scold me too much.

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by SajanGhimire1** [comment, marked as answer]
@qSHiFTp 

Changed ```fname.classList.toggle(classname);``` to ```elements[i].classList.toggle(classname);```in the allCl function
```
let body = document.querySelector('body');
let themeButton = document.querySelector('.button_2');
let button1 = document.querySelectorAll('.button_1');
let text = document.querySelectorAll('.text');

function allCl (elements, classname) {
  for (let i = 0; i < elements.length ; i++) {
    elements[i].classList.toggle(classname); 
    console.log(classname);
  }
}

const darkTheme = window.matchMedia("(prefers-color-scheme: dark)");
if (darkTheme.matches) {
  body.classList.toggle('dark-theme'); 
  allCl(button1, 'button_1');
  allCl(text, 'dark-text');
} else {
  body.classList.toggle('light-theme'); 
  allCl(button1, 'button_1');
  allCl(text, 'text');
}

themeButton.onclick = function () {
  body.classList.toggle('light-theme');
  body.classList.toggle('dark-theme');
  allCl(button1, 'button_1');
  allCl(button1, 'dark-button_1');
  allCl(text, 'text');
  allCl(text, 'dark-text');
}
```

**Reply by qSHiFTp** [reply]
Thank you
